account-compression/sdk/tests/accountCompression.test.ts

```diff
@@ -0,0 +1,684 @@
+          return treeNode.node;
+        })
+      );
+      assert(
+        replaceLeafIx.keys.length == 3 + MAX_DEPTH,
+        `Failed to create proof for ${MAX_DEPTH}`
+      );
+
+      await execute(provider, [replaceLeafIx], [payer]);
+
+      updateTree(offChainTree, newLeaf, index);
+
+      const splCMT = await getConcurrentMerkleTree(connection, splCMTKeypair.publicKey);
+      const onChainRoot = getCMTCurrentRoot(splCMT);
+
+      assert(
+        Buffer.from(onChainRoot).equals(offChainTree.root),
+        "Updated on chain root matches root of updated off chain tree"
+      );
+    });
+
+    it("Replace that leaf with a minimal proof", async () => {
+      const previousLeaf = offChainTree.leaves[0].node;
+      const newLeaf = crypto.randomBytes(32);
+      const index = 0;
+
+      const replaceLeafIx = createReplaceIx(
+        payer,
+        splCMTKeypair.publicKey,
+        offChainTree.root,
+        previousLeaf,
+        newLeaf,
+        index,
+        getProofOfLeaf(offChainTree, index, true, 1).map((treeNode) => {
+          return treeNode.node;
+        })
+      );
+      assert(
+        replaceLeafIx.keys.length == 3 + 1,
+        "Failed to minimize proof to expected size of 1"
+      );
+      await execute(provider, [replaceLeafIx], [payer]);
+
+      updateTree(offChainTree, newLeaf, index);
+
+      const splCMT = await getConcurrentMerkleTree(connection, splCMTKeypair.publicKey);
+      const onChainRoot = getCMTCurrentRoot(splCMT);
+
+      assert(
+        Buffer.from(onChainRoot).equals(offChainTree.root),
+        "Updated on chain root matches root of updated off chain tree"
+      );
+    });
+  });
+
+  describe("Examples tranferring appendAuthority", () => {
+    const authority = Keypair.generate();
+    const randomSigner = Keypair.generate();
+    describe("Examples transferring authority", () => {
+      it("... initializing tree ...", async () => {
+        await provider.connection.confirmTransaction(
+          await (connection as Connection).requestAirdrop(
+            authority.publicKey,
+            1e10
+          )
+        );
+        [splCMTKeypair, offChainTree] = await createTreeOnChain(
+          authority,
+          1
+        );
+      });
+      it("Can transfer authority", async () => {
+        const transferAuthorityIx = createTransferAuthorityIx(
+          authority,
+          splCMTKeypair.publicKey,
+          randomSigner.publicKey
+        );
+        await execute(provider, [transferAuthorityIx], [authority]);
+
+        const splCMT = await getConcurrentMerkleTree(connection, splCMTKeypair.publicKey);
+
+        assert(
+          getCMTAuthority(splCMT).equals(randomSigner.publicKey),
+          `Upon transfering authority, authority should be ${randomSigner.publicKey.toString()}, but was instead updated to ${getCMTAuthority(splCMT)}`
+        );
+      });
+      it("Attempting to replace with new authority now works", async () => {
```

Comment:
This seems to be testing that the random signer doesn't work, not that the new authority works

Reply:
Great catch -- separated this into two separate tests, as was initially intended